neqo-http3-server/src/main.rs

```diff
@@ -92,14 +92,17 @@ fn process_events(server: &mut Http3Server) {
 
                 let default_ret = b"Hello World".to_vec();
 
-                let response = match headers.iter().find(|&(k, _)| k == ":path") {
-                    Some((_, path)) if !path.is_empty() => {
-                        match path.trim_matches(|p| p == '/').parse::<usize>() {
-                            Ok(v) => vec![b'a'; v],
-                            Err(_) => default_ret,
+                let response = match headers {
```

Comment:
Under what conditions do we allow headers to be None, as opposed to an empty set.

Also, we should reject anything without ':path' to keep this simple.

Reply:
If stream gets closed before headers we return None. If we get empty header frame we return empty set.
We probably should just ignore streams being closed without a header frame, I have not fix this yet, because there are some other small corner cases that I wanted to do at once (they are all about how much semantics neqo-htttp3 implements and what it lefts for the application)

for neqo-http3-server we should reject anything without a :path.